watertap/property_models/coagulation_prop_pack.py

```diff
@@ -0,0 +1,748 @@
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Initial property package for water treatment via a Coagulation-Flocculation process
+"""
+from pyomo.environ import (Constraint,
+                           Var,
+                           Param,
+                           Expression,
+                           Reals,
+                           NonNegativeReals,
+                           Suffix,
+                           value,
+                           assert_optimal_termination,
+                           check_optimal_termination)
+from pyomo.environ import units as pyunits
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        MaterialFlowBasis,
+                        PhysicalParameterBlock,
+                        StateBlockData,
+                        StateBlock,
+                        MaterialBalanceType,
+                        EnergyBalanceType)
+from idaes.core.components import Component
+from idaes.core.phases import LiquidPhase, SolidPhase, PhaseType
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars, solve_indexed_blocks
+from idaes.core.util.model_statistics import degrees_of_freedom, number_unfixed_variables
+from idaes.core.util.exceptions import PropertyPackageError, InitializationError, ConfigurationError
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+from idaes.core.util import get_solver
+
+__author__ = "Austin Ladshaw"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+# Forward declaration of 'CoagulationParameterData'
+@declare_process_block_class("CoagulationParameterBlock")
+class CoagulationParameterData(PhysicalParameterBlock):
+    CONFIG = PhysicalParameterBlock.CONFIG()
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super(CoagulationParameterData, self).build()
+
+        self._state_block_class = CoagulationStateBlock
+
+        # phases
+        self.Liq = LiquidPhase()
+        self.Sol = SolidPhase()
+
+        # components
+        self.H2O = Component(default={"valid_phase_types": PhaseType.liquidPhase})
+        self.TSS = Component(default={"valid_phase_types": PhaseType.liquidPhase})
+        self.TDS = Component(default={"valid_phase_types": PhaseType.liquidPhase})
+        self.Sludge = Component(default={"valid_phase_types": PhaseType.solidPhase})
+
+        # ====== parameters ========
+        mw_comp_data = {'H2O': 18.01528e-3,
+                        'TSS': 31.4038218e-3,
+                        'TDS': 31.4038218e-3,
+                        'Sludge': 31.4038218e-3}
```

Comment:
What are the estimates for molecular weight values of TSS, TDS, and Sludge based on?

Reply:
So, there is no such thing as a MW for TSS and Sludge, but I wanted to have something here just in case we needed molar flows (for whatever reason). In this case, I just assigned TSS and Sludge to have same MW as TDS (from the seawater prop pack). 

The mw_comp param is Mutable, so anyone can change these values if necessary.

Reply:
I wonder if it would make sense to only apply molar flows for H2O and TDS and omit for TSS and Sludge. Otherwise, if someone tries to get the molar flows for all components, they are going to see strange results for TSS and sludge that aren't really applicable.